### ISSUE TITLE
Keyboard shortcuts wont work

### ISSUE THREAD

**Issue by BroloPie66** [photopea/photopea#948]
I have tried this in multiple browsers and nothing works.

**Comment by curryarts** [comment]
I am having the same problem - no keyboard response. No response when I try ALT & CTL, and no shortcuts work.

**Comment by curryarts** [comment]
The keyboard doesn't respond in Edge, but it does work in Chrome and Firefox.
So, it looks like the problem is with Edge.

**Comment by photopea** [comment]
Oh yes, right. MS Edge does not support the new standard KeyboardEvent.code, see here: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/18367767/ . There is my last comment on that side about Photopea. I recommend you to use some better browser, until they fix it.

**Comment by XamarinDeveloper** [comment]
Hey, on Ubuntu, on Firefox and Chrome
The Ctrl shortcuts aren't working as well as Shift + Ctrl although I was able to create a new file with Alt + Ctrl and weird enough I was able to select all by Alt + Ctrl + A combination
and Ctrl shortcuts actually behavior like there is no Ctrl and there is just the letter for example Ctrl + A selects the Path Select tool

**Comment by photopea** [comment]
Go to this demo, click into white area and press keys: http://jsfiddle.net/p45f44zb/14/  It should show keys, which are pressed at the moment. What do you see?

I guess that in your case, the bug is in a browser or in the OS.

**Comment by XamarinDeveloper** [comment]
Thank you for the fast response :)
If I press Ctrl first it ignores it, but if I press Shift or Alt first it gets the Ctrl as well
And Alt + Shift doesn't work either way
But I know that Ctrl shortcuts work in Chrome so this means that the bug is in Chrome probably
I'll take it to them then (not sure if it's ever gonna be solved)

**Comment by photopea** [comment]
So everything is working now or it still does not? Can you update your browser, your OS, or maybe your bios? Since Photopea works for millions of users with different environments, I think the problem is in your device.

**Comment by XamarinDeveloper** [comment]
No, the code I've linked at the end of my last comment is slightly different from your code, and it works.
I've tested it on a totally different system and the shortcuts WORK but only the laptop models are different and my Chrome is 2 versions ahead of his (maybe the bug is added in my version 🙇)
Anyway, in my system, the Ctrl keydown event isn't sent individually and should be gotten in another key's keydown events by checking `e.ctrlKey` properties, I mean, I know your system now works perfectly, but it'd be cool if you added the type of Ctrl support that my system works in, even if you don't, I totally understand and thank you for your time

**Comment by photopea** [comment]
We do not guarantee the correct functionality of Photopea in experimental
or other non-standard browsers. I think it is not correct to make our code
larger just to overcome bugs in browsers.

sobota 11. července 2020 Xamarin-Developer <notifications@github.com>
napsal(a):

**Comment by rareseu** [comment]
> My Cntl+[*] key shortcut started working when I turned off my OS's (Linux Mint 20.04) option to show "position of pointer when Control key is pressed".

Can confirm what @toyclyde says. I noticed that when the "Show position of pointer when the Control key is pressed" option is enabled in Mouse settings (Linux Mint 22.1 Cinnamon), common CTRL shortcuts like CTRL+J, CTRL+Z, etc., do not work in many web apps such as Canva and Photopea. Disabling this setting restores normal shortcut functionality.
